tests/test_path.rs

```diff
@@ -50,3 +50,17 @@ fn parse_interpolated_leading_component() {
     }
     "###);
 }
+
+#[test]
+fn print_incomplete_qpath() {
+    use quote::ToTokens;
+    use syn::Type;
+
+    let mut ty: Type = syn::parse_quote!(<Self as A>::Q);
+    if let Type::Path(tp) = &mut ty {
+        tp.path.segments.pop();
+        tp.to_token_stream();
+        tp.path.segments.pop();
+    }
+    ty.to_token_stream();
```

Comment:
btw, the output of first `.to_token_stream()` is currently broken (second is printed as `<Self>`)
: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e063b5e4bb31119e2bc57cae3d7cdc22

I think it might be preferable to print it like `<Self as A>` (because we can write like `quote!(#tp::#path_assoc_item)`), but I haven't looked into its effect on others, so I don't touch it in this PR.

Reply:
Please update both of these to_token_stream calls to compare the return value against what is expected, and update the ToTokens impl to produce something reasonably useful for the cases that we care about supporting.